releasenotes/notes/time-evo-framework-9d58827bdbbebd62.yaml

```diff
@@ -0,0 +1,14 @@
+---
+features:
+  - |
+    Interfaces for the unified framework for Quantum Time Evolution are introduced.
+    :class:`~qiskit.algorithms.evolvers.EvolutionProblem` defines an evolution problem and
+    can be passed to any evolution algorithm available in the framework.
+    :class:`~qiskit.algorithms.evolvers.Evolver` interface serves as a base for
+    any time evolution algorithm for evolving quantum states, including evolutions based on
+    time-dependent Hamiltonians.
+    :class:`~qiskit.algorithms.evolvers.imaginary.ImaginaryEvolver` and
+    :class:`~qiskit.algorithms.evolvers.real.RealEvolver` are derived interfaces for
+    imaginary and real time evolution cases respectively.
+    :class:`~qiskit.algorithms.evolvers.EvolutionResult` is introduced as a result object
```

Comment:
Can you check what happens here - we import and  document these at the algorithms level, so in my mind these would preferably be links to the classes at that level. I am not sure whether these links will resolve. For now we chose to keep Evolver more private so to speak so its not exported, rather the Real and Imaginary sub-classes thereof  are, so I think the Evolver one will fail.